### DISCUSSION TITLE
Shouldn't URL of the dataset also be included?

### DISCUSSION THREAD

**Discussion by entin-hun** [mapping-commons/sssom#551] [Q&A]
Dear All, Please answer my noob question:
I'm looking for a solution that is between native integrations and a documented API: an easy-to-setup mapping of fields (stored in any accessible database) for our API. I saw gr8 potential in SSSOM but then I realized that it might just contextualize fields but I don't see where the URL of the dataset is specified.
Is this because it isn't ideal for such use case or the recommended usage is storing those info beneath the SSSOM.tsv file? (credentials, OpenAPI GET endpoint or do you know about such a schema?)
Thanks for clarifying!🙏🏽

**Comment by matentzn** [comment]
@entin-hun thank you for reaching out! Before I can help you answering all your questions I need to ask you some things.

> I realized that it might just contextualize fields but I don't see where the URL of the dataset is specified.

The URL of the dataset is usually specified with [subject_source](https://mapping-commons.github.io/sssom/subject_source/) and `object_source`. This field is essentially saying: "the subject of the mapping lives in X". Is this what you mean here? I would be great if you could can provide a full example of your use case, ideally in some kind of table format.

> Is this because it isn't ideal for such use case or the recommended usage is storing those info beneath the SSSOM.tsv file? (credentials, OpenAPI GET endpoint or do you know about such a schema?)

I need to know a concrete example here - You can always share your API credentials either using a [extension](https://github.com/mapping-commons/sssom/issues/328), or as part of the "comment" field if needed. 

In any case, I think I am misunderstanding the question so lets continue with a concrete example :)

**Comment by entin-hun** [comment]
Thank you @matentzn for replying quickly and mindfully! Context of our use-case:
SSSOM or LinkML seems to be an appropriate schema for data providers to guide GET requests which fields to fetch. When someone wants to utilize already existing data (but needs only a subset or 5 fields from the 35), there are still cases when the data provider agrees but can't/doesn't want to send POST requests to the former (e.g. SQL or if it's a manually-updated Excel).
In these cases the utilizers could make the GET requests, based on data providers' field schema. What do you think?

**Comment by matentzn** [comment]
@entin-hun I am very sorry, I think you caught me in a scenario very far outside my comfort zone!

> SSSOM or LinkML seems to be an appropriate schema for data providers to guide GET requests which fields to fetch.

This sentence alone is difficult for me to understand as I do not know how you generally "guide" someone to the right fields to fetch. 

Unfortunately I dont understand your use case at all. Did you have better luck asking the LinkML folks? Else, maybe I need a more fleshed out concrete example which explains what role the mappings play. Sorry about that! Please keep asking until we nail this down.
